console/src/components/Services/Data/DataSubSidebar.js

```diff
@@ -236,66 +236,41 @@ const DataSubSidebar = props => {
         setTreeViewItems(newItems);
       }
     );
-  }, [sources.length, tables, functions, enums, schemaList, currentTable]);
+  }, [sources.length, tables, functions, enums, schemaList, currentTable, allSourcesSchemas, sources, dispatch, getItems, isFetching, preLoadState]);
```

Comment:
This probably will cause the problems with rerenders; but fix it properly by making these objects memoised instead of not adding them to the dep list.